pkg/eks/components/CruEKS.vue

```diff
@@ -0,0 +1,778 @@
+        if (out[type.groupLabel]) {
+          out[type.groupLabel].push(type);
+        } else {
+          out[type.groupLabel] = [type];
+        }
+      });
+
+      return out;
+    },
+
+    instanceTypeOptions(): AWS.InstanceTypeOption[] {
+      const out: AWS.InstanceTypeOption[] = [];
+
+      Object.keys(this.groupedInstanceTypes).forEach((groupLabel: string) => {
+        const instances = this.groupedInstanceTypes[groupLabel];
+        const groupOption = { label: groupLabel, kind: 'group' };
+        const instanceTypeOptions = instances.map((instance: AWS.InstanceType) => {
+          return {
+            value: instance.apiName,
+            label: instance.label,
+            group: instance.groupLabel
+          };
+        });
+
+        out.push(groupOption);
+        out.push(...instanceTypeOptions);
+      });
+
+      return out;
+    },
+
+    spotInstanceTypeOptions(): AWS.InstanceTypeOption[] {
+      const out: AWS.InstanceTypeOption[] = [];
+
+      Object.keys(this.groupedInstanceTypes).forEach((groupLabel: string) => {
+        const instances = this.groupedInstanceTypes[groupLabel];
+        const groupOption = { label: groupLabel, kind: 'group' };
+        const instanceTypeOptions = instances.reduce((spotInstances: AWS.InstanceTypeOption[], instance: AWS.InstanceType) => {
+          if (!(instance.supportedUsageClasses || []).includes('spot')) {
+            return spotInstances;
+          }
+          const opt = {
+            value: instance.apiName,
+            label: instance.label,
+            group: instance.groupLabel
+          };
+
+          spotInstances.push(opt);
+
+          return spotInstances;
+        }, []);
+
+        if (instanceTypeOptions.length) {
+          out.push(groupOption);
+          out.push(...instanceTypeOptions);
+        }
+      });
+
+      return out;
+    }
+  },
+
+  methods: {
+    setClusterName(name: string): void {
+      this.$set(this.normanCluster, 'name', name);
+      this.$set(this.config, 'displayName', name);
+    },
+
+    onMembershipUpdate(update: {newBindings: any[], removedBindings: any[], save: Function}): void {
+      this.$set(this, 'membershipUpdate', update);
+    },
+
+    async saveRoleBindings(): Promise<void> {
+      if (this.membershipUpdate.save) {
+        await this.membershipUpdate.save(this.normanCluster.id);
+      }
+    },
+
+    // only save values that differ from upstream aks spec - see diffUpstreamSpec comments for details
+    removeUnchangedConfigFields(): void {
+      const upstreamConfig = this.normanCluster?.eksStatus?.upstreamSpec;
+
+      if (upstreamConfig) {
+        const diff = diffUpstreamSpec(upstreamConfig, this.config);
+
+        this.$set(this.normanCluster, 'eksConfig', diff);
+      }
+    },
+
+    async actuallySave(): Promise<void> {
+      await this.normanCluster.save();
+
+      return await this.normanCluster.waitForCondition('InitialRolesPopulated');
```

Comment:
this one kicks off a TS error. given where `waitForCondition` comes from i'm not sure we can avoid anything other than an `as` here

Reply:
I don't see this error. Different vscode configuration? I'm using the official vue extension as well as eslint and prettier, and set vscode to use the workspace's ts version (v4.5.5)
![Screenshot 2024-04-08 at 10 08 58 AM](https://github.com/rancher/dashboard/assets/42977925/9e4bea3e-b0cf-4ea3-8cf3-b59c41f227f2)

Reply:
ah, it's a vetur error (i always struggled with the official extension). It's picking up that the return value for `waitForCondition` is `Promise<unknown>` which  contradicts the fn's `Promise<void>`.

![image](https://github.com/rancher/dashboard/assets/18697775/910a0a61-122d-4b73-9d56-4ea675f46ed1)

can we update the return object to match?

Reply:
Frustratingly I couldn't get this error to show up even switcher over to vetur. I've typed the return value for `waitForCondition` now though